### PR TITLE
Integrate spdlog logger system

### PR DESCRIPTION
**Pull request by bryanperris**
This pull request integrates the spdlog logging system into Descent3 to modernize and enhance our logging capabilities. The integration aims to replace the existing logging system gradually, improve logging performance, and increase the detail of diagnostic data captured. This change sets the stage for more detailed and efficient debugging and monitoring throughout the game's development lifecycle.

Note: I have only setup the logger for the Linux side of things, Windows/MacOS will have to follow suit.

### CODE REVIEW

File: .github/workflows/build.yml
Comment (winterheart) [inline .github/workflows/build.yml:77]: You don't need that in pipeline. Just add spdlog to vcpkg.json

Comment (bryanperris) [reply .github/workflows/build.yml]: yeah sorry, should be all set now, all checks pass.

**Review by winterheart — APPROVED** [review]
OK for first iteration.

File: Descent3/loki_utils.c
Comment (Lgt2x) [inline Descent3/loki_utils.c:478]: delete line

File: Descent3/winmain.cpp
Comment (Lgt2x) [inline Descent3/winmain.cpp:57]: revert encoding change

**Review by Lgt2x — COMMENTED** [review]
Please rebase&squash commits before merging

**Comment by bryanperris** [comment]
> Please rebase&squash commits before merging

Sure, and btw it has been rebased already.

File: Descent3/winmain.cpp
Comment (bryanperris) [reply Descent3/winmain.cpp:57]: I don't see that diff locally.

**Comment by Lgt2x** [comment]
> > Please rebase&squash commits before merging
> 
> Sure, and btw it has been rebased already.

I was talking about squashing commits in interactive rebase mode `git rebase -i main`, you seem to have a lot of development commits that should not appear in the main branch once merged. Another option is using the "squash commits" checkbox when merging on github, but I prefer the former.

File: Descent3/winmain.cpp
Comment (Lgt2x) [reply Descent3/winmain.cpp:57]: your IDE changed encoding from WINDOWS-1252 to UTF-8

**Review by Lgt2x — APPROVED** [review]

**Comment by Lgt2x** [comment]
merging when encoding is fixed

**Comment by bryanperris** [comment]
> merging when encoding is fixed

I couldn't figure out why it happens, maybe a git thing.  My local diff doesn't show changes made to it, only I can find them in the diff on GitHub.

File: Descent3/winmain.cpp
Comment (winterheart) [reply Descent3/winmain.cpp:57]: We should decide what to do with these messages. Last time I mentioned that changing encoding to UTF-8 may broke messages on locales other than en, but now I think we can just convert these files into UTF-8 and deal with potential issues with non-English locales later.

Comment (bryanperris) [reply Descent3/winmain.cpp:57]: It would be a lot nicer to use GNU gettext or something so the languages lives in separate files, and not the source files.

Comment (winterheart) [reply Descent3/winmain.cpp:57]: You can rebase onto current main, and this encoding mismatch will be gone.

Comment (bryanperris) [reply Descent3/winmain.cpp:57]: It turns out, it was vs code to blame here, I found it locally.

**Pull request merged**
